Source/WarpX.cpp

```diff
@@ -1552,7 +1553,7 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
         deposit_charge = do_dive_cleaning || update_with_rho || current_correction;
     }
-    if (deposit_charge)
+    if (!external_solver && deposit_charge)
```

Comment:
I don't think this approach would work since we would still need `rho_fp` initialized when using either the MLMG or direct solver. There should be another if statement for `external_solver` around the parts that initialize `full_rho_fp` (lines 1561 - 1573 below).